### PR TITLE
Alerting: update panel alert tab

### PR DESCRIPTION
**Pull request by domasx2**
**What this PR does / why we need it**:

* Reworked pamel alerts tab for unified alerting. Shows alert rules that match the panel either by dashboard & panel id annotations or by panel URL annotation. Has a button that takes you to rule form pre-filled with queries, rule name and folder from the panel
* Possible to provide default values to rule form via URL `defaultValues` query param. Expects JSON with form values
* Possible to specify where the rule form should return to on "Cancel" or "Save and exit" with `returnTo` query param. 
* More suggested annotation options
* Alert threshold not shown on the graph for unified alerting. Currently don't have an easy way to calculate this threshold. Will liekly be revisited later
* Condition field in rule form automatically selects the last expression if one exists
* Some annotations in rule/alert details show "readable" name with tooltip for the real key. This is somewhat odd, might need more thinking about in the future. Real key is important as that's what you'd use in a template

TODO:
- [x] remove the `dasboard_url` and `panel_url` annotations, these are to be generated at runtime. Not quite sure how to link existing rules to panels though.. perhaps that's not needed

https://user-images.githubusercontent.com/847684/117804340-55bd3580-b260-11eb-93f3-444b8f04e22a.mp4

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/alerting-squad/issues/9
Fixes https://github.com/grafana/alerting-squad/issues/141
Fixes https://github.com/grafana/alerting-squad/issues/110

**Special notes for your reviewer**:

I went with an odd way to get the number of rules into panel alerting tab. Better ideas are welcome!

Does anything else need to be modified?

### CODE REVIEW

**Review by nathanrodman — APPROVED** [review]
Great work! It's pretty seemless going from panel to alert rule creation, and the navigation back to the panel from the rule list is awesome.

File: public/app/features/alerting/unified/PanelAlertTabContent.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          </div>
+        </CustomScrollbar>
+      </>
+    );
+  }
+
+  return (
+    <div className={styles.noRulesWrapper}>
+      {alert}
+      {!!dashboard.uid && (
```
Comment (bfmatei) [inline public/app/features/alerting/unified/PanelAlertTabContent.tsx:57]: Wouldn't this work better as a ternary operator?

File: public/app/features/alerting/unified/PanelAlertTabContent.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      <div className={styles.innerWrapper}>
+        {alert}
+        <LoadingPlaceholder text="Loading rules..." />
+      </div>
+    );
+  }
+
+  if (rules.length) {
+    return (
+      <>
```
Comment (bfmatei) [inline public/app/features/alerting/unified/PanelAlertTabContent.tsx:42]: Is this fragment necessary?

**Review by bfmatei — APPROVED** [review]
Just some super minor nitpicks

**Pull request merged**